test/unit/test_rewrite_arithmetic.py

```diff
@@ -125,8 +125,12 @@ def test_rewrite_arithmetic_frequency():
     response_program = Program(response.quil)
     assert response_program[0].out() == "DECLARE __P1 REAL[2]"
     assert response_program[1].out() == "DECLARE theta REAL[1]"
-    assert response_program[2] == fdefn0
-    assert response_program[3] == fdefn1
+    assert (
+        response_program[2] == fdefn0
+        and response_program[3] == fdefn1
+        or response_program[2] == fdefn1
+        and response_program[3] == fdefn0
+    )
```

Comment:
This isn't directly related to these changes, but I fixed it here since I noticed it was flaky. The order of frame definitions in a printed `Program` isn't guaranteed by `quil-rs` since they are stored in a `HashMap`. `DefFrame` also doesn't support `Hash` because they contain a `HashMap` field, leading to this verbose assertion.

Reply:
Would appreciate parentheses for those who don't have operator precedence memorized

```python
    assert (
        (
            response_program[2] == fdefn0
            and response_program[3] == fdefn1
        ) or (
            response_program[2] == fdefn1
            and response_program[3] == fdefn0
        )
    )
```

Alternatively, create a set and compare it?

```python
    assert ({ response_program[2], response_program[3] } == { fdefn0, fdefn1 })
```

Reply:
I added the parentheses (with different formatting, because black). Using a `set` would be my preferred method, but a `DefFrame` can't be hashed, so it can't be put in a `set`.